### PR TITLE
FIX: body[:comment_content] can be nil and strip can't be used

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: lib/akismet.rb [hunk 1]
```diff
@@ -71,7 +71,8 @@ def self.user_agent_string
     def post(path, body)
       # Send a maximum of 32000 chars which is the default for
       # maximum post length site settings.
-      body[:comment_content] = body[:comment_content].strip[0..31999]
+      comment_content = body[:comment_content] || ""
+      body[:comment_content] = comment_content.strip[0..31999]
```
Comment (eviltrout) [inline lib/akismet.rb:75]: It's insignificant really but you could do this: `body[:comment_content] = (body[:comment_content] || "").strip[0..31999]`

**Review by eviltrout — APPROVED** [review]

**Review by romanrizzi — CHANGES_REQUESTED** [review]
Sorry for not jumping in earlier. 

As we discussed on chat, this will send Akismet an empty body. We should add a `nil` check here:

https://github.com/discourse/discourse-akismet/blob/master/lib/discourse_akismet/users_bouncer.rb#L16

**Comment by eviltrout** [comment]
I'll close this as the wrong approach then. @romanrizzi can you handle the nil check as a follow up PR?

**Comment by ZogStriP** [comment]
Fixed in https://github.com/discourse/discourse-akismet/commit/1e7d31eb76f5fde46cefc06ce54026911c769e2f.

**Pull request closed without merging**